demos/common/processing/input_data.h

```diff
@@ -0,0 +1,47 @@
+/*
+// Copyright (C) 2018-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#pragma once
+#include <samples/ocv_common.hpp>
+
+struct InputData {
+    virtual ~InputData() {}
+
+    template<class T> T* asPtr() {
```

Comment:
I don't understand the purpose of this. You could achieve the same effect with `dynamic_cast` on a reference.

Reply:
it checks for proper cast result and throws exception in case of incorrect cast. Besides, it provides more compact way to cast.

Reply:
I've simplified code of those functions - thank you for the hint. But I still want to keep them as they simplify casting inside other code. Also, it might be useful to keep them so we could replace dynamic cast to static cast later (to save some time on runtime checks in release version for instance).